From: Casey Bodley <cbodley@redhat.com>
Date: Mon, 16 May 2016 09:37:05 -0400
Subject: radosgw-admin: fix 'period push' handling of --url

was calling send_to_remote_gateway(), but passing 'url' instead of
'remote'. now uses send_to_remote_or_url() to accept either

Fixes: http://tracker.ceph.com/issues/15926

Signed-off-by: Casey Bodley <cbodley@redhat.com>
(cherry picked from commit e885f1e3be0f9a0c6457c5387f1c38b6e8b6a492)

Resolves: rhbz#1336648
---
 src/rgw/rgw_admin.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/rgw/rgw_admin.cc b/src/rgw/rgw_admin.cc
index 657adcd..87c27ed 100644
--- a/src/rgw/rgw_admin.cc
+++ b/src/rgw/rgw_admin.cc
@@ -3819,7 +3819,8 @@ int main(int argc, char **argv)
       jf.flush(bl);
 
       JSONParser p;
-      ret = send_to_remote_gateway(url, info, bl, p);
+      ret = send_to_remote_or_url(remote, url, access_key, secret_key,
+                                  info, bl, p);
       if (ret < 0) {
         cerr << "request failed: " << cpp_strerror(-ret) << std::endl;
         return ret;
